download_metadata.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	log "github.com/sirupsen/logrus"
+)
+
+func metadataDownloader(Metadataclient *MetadataBackend) {
+
+	var (
+		Bucket         = Metadataclient.Bucket // Download from this bucket
+		Prefix         = "datasets/"           // Using this key prefix
+		LocalDirectory = "web/content/"        // Into this directory
+	)
+	client := Metadataclient.Client
+
+	manager := manager.NewDownloader(client)
+	paginator := s3.NewListObjectsV2Paginator(client, &s3.ListObjectsV2Input{
+		Bucket: &Bucket,
+		Prefix: &Prefix,
+	})
+	log.Infoln("Started downloading metadata files")
+	for paginator.HasMorePages() {
+		page, err := paginator.NextPage(context.TODO())
+		if err != nil {
+			log.Fatalln("error:", err)
```

Comment:
```suggestion
			log.Fatalf("Error while getting next expected page from ListObjectsV2Paginator: %v", err)
```

Reply:
(My preference for bringing in the formatter, more informative message of what goes wrong.

Reply:
changed it